### PR TITLE
IGNITE-23016 Fixed to prevent implicit use of the direct-io on a dev branch

### PR DESCRIPTION
**Pull request by okreda1**
IGNITE-23016 ducktape tests for dev branch implicitly use direct-io
- Ignored the direct-io module through EXCLUDE_MODULES env, which checks if direct-io is present in modules
- Updated envs method in IgniteNodeSpecExcludeDucktests to accommodate changes

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py [hunk 1]
```diff
@@ -245,12 +245,16 @@ def envs(self):
...
         :return: environment set.
         """
-        return {
+        environment_dict = {
             'EXCLUDE_TEST_CLASSES': 'true',
             'IGNITE_LOG_DIR': self.service.log_dir,
             'USER_LIBS': ":".join(self.libs()),
             "MAIN_CLASS": self.service.main_java_class
         }
+        if "direct-io" not in self.modules():
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:254]: Please  follow here the "Semantic Units" rule from the coding guidelines:
https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-SemanticUnits

It's for java but general requirements are also need to be followed in python code as well.

So, add  empty line before this line.

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py [hunk 2]
```diff
@@ -245,12 +245,16 @@ def envs(self):
...
+            # IGNITE-23016, excluded to avoid implicit usage
+            environment_dict['EXCLUDE_MODULES'] = "direct-io"
+        return environment_dict
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:257]: Add  empty line before this one.

File: modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py [hunk 3]
```diff
@@ -89,7 +89,9 @@ def envs(self):
         Skip the module target directory while building classpath.
         """
         envs = super().envs()
-
-        envs["EXCLUDE_MODULES"] = "ducktests"
+        if envs.get("EXCLUDE_MODULES") is not None:
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py:92]: Add  empty line before this one.

**Review by skorotkov — CHANGES_REQUESTED** [review]

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py (hunk 2 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:259]: added blank line

File: modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py (hunk 3 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py:93]: Added blank line.

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py (hunk 1 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:255]: Fixed style.

File: modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py [hunk 4]
```diff
@@ -35,6 +34,7 @@ class ThinClientQueryTest(IgniteTest):
     to use with ssl enabled:
     export GLOBALS='{"ssl":{"enabled":true}}' .
     """
+
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py:37]: Please remove this unintended change

File: modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py [hunk 5]
```diff
@@ -70,26 +70,33 @@ def test_thin_client_index_query(self, server_version, filter):
         ignite.stop()
 
 
-class IgniteNodeSpecExcludeDucktests(IgniteNodeSpec):
-    """
-    Ignite node specification that excludes module 'ducktests' from classpath.
-    """
-    def modules(self):
+def return_spec_without_ducktests(service, base_spec):
```
Comment (skorotkov) [inline modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py:73]: Please add description of the function and parameters

**Review by skorotkov — CHANGES_REQUESTED** [review]

File: modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py (hunk 5 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py:72]: done

File: modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py (hunk 4 above)
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py]: fixed

**Review by skorotkov — APPROVED** [review]

**Review by skorotkov — CHANGES_REQUESTED** [review]
Please change the PR title to indicate what exactly was done.

**Review by skorotkov — APPROVED** [review]

**Review by timoninmaxim — CHANGES_REQUESTED** [review]

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py [hunk 6]
```diff
@@ -245,13 +245,19 @@ def envs(self):
...
-        return {
+        environment_dict = {
             'EXCLUDE_TEST_CLASSES': 'true',
             'IGNITE_LOG_DIR': self.service.log_dir,
             'USER_LIBS': ":".join(self.libs()),
             "MAIN_CLASS": self.service.main_java_class
         }
 
+        if "direct-io" not in self.modules():
+            # IGNITE-23016, excluded to avoid implicit usage
```
[the remark this answers is not in the document]
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py]: Done

File: modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py [hunk 7]
```diff
@@ -21,7 +21,6 @@
 from ignitetest.services.ignite import IgniteService
 from ignitetest.services.ignite_app import IgniteApplicationService
 from ignitetest.services.utils.ignite_configuration import IgniteConfiguration, IgniteThinClientConfiguration
-from ignitetest.services.utils.ignite_spec import IgniteNodeSpec
```
[the remark this answers is not in the document]
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py]: Done

[the remark this answers is not in the document]
Comment (okreda1) [reply modules/ducktests/tests/ignitetest/tests/thin_client_query_test.py]: https://github.com/apache/ignite/pull/11594

**Review by timoninmaxim — APPROVED** [review]

**Pull request merged**